src/lib.rs

```diff
@@ -1334,11 +1335,12 @@ impl Build {
                     .print_metadata(&format_args!("cargo:rustc-link-lib={}", stdlib.display()));
             }
             // Link c++ lib from WASI sysroot
-            if self.get_target()?.contains("wasi") {
+            if Build::is_wasi_target(target.as_ref()) {
                 let wasi_sysroot = self.wasi_sysroot()?;
                 self.cargo_output.print_metadata(&format_args!(
-                    "cargo:rustc-flags=-L {}/lib/wasm32-wasi -lstatic=c++ -lstatic=c++abi",
-                    Path::new(&wasi_sysroot).display()
+                    "cargo:rustc-flags=-L {}/lib/{} -lstatic=c++ -lstatic=c++abi",
```

Comment:
Can you explain a bit about this change please?

In your previous PR it was hard coded, and now it changes to wasm32-wasi

Reply:
Hmmm maybe I'm just thinking too much, it's probably because we expanded the targets supported.

Reply:
Yes, the lib folder structure in sysroot looks like this: ![image](https://github.com/rust-lang/cc-rs/assets/7198769/41d0edbd-4254-4b23-aea9-aa2057c43ce4)